### DISCUSSION TITLE
`<part-name-display>` within `<print>`

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#570] [Q&A] [answered]
The standard allows both `<part-name-display>` and `<part-abbreviation-display>` to be children of `<print>`.

Can someone please give me a real-world example for the former?  I currently can't think of a situation where the full instrument name appears within a part (at the left side of a staff) except at the very beginning.

**Comment by PhysNovalOnGit** [comment]
There are some anthologies and collections for children (and teens) that have both _solo_ pieces (or a monophonic instrument (such as flute or violin) + accompanement) and _ensembles_ in them. In such a case the full instrument name can appear midbook. I have some of these marked this way.
This is how the upper halves of the pages no. 55 and 69 of [this book](https://www.musica.ru/product/albom-balalaechnika-vyp-1-sost-gorbachyov-a-inshakov-i?lang=en) look like:
![GavotGossek](https://github.com/user-attachments/assets/268e276d-5555-4fc5-928f-0d52e6091890)

![TriZverolova](https://github.com/user-attachments/assets/140769ea-80ae-4d96-93a3-ffc58e060e53) 
Ensembles start on the page 55. There are 6 solo pieces with accompanement before Gossec's Gavotte. 
Фортепиано is a full name and Ф.-п. is an abbreviated one. It seems customary that every stave in the beginning of an ensemble is fully marked (unless the name is uncomfortably long (Kontrabass, Фортепиано - 10 letters) and abbreviating it allows to fit a 4 bar phrase into the textwidth.) At least it is customary in Russian books.

**Reply by lemzwerg** [reply]
Thanks!  Unfortunately, this doesn't answer my question.  Reason is that most certainly the two pieces you mention are *separate parts* in the MusicXML sense.  For such situations it is clear how to display the full instrument name at the beginning if desired.

What I want to know is an example of a *single* part (in the MusicXML sense) where you are going to display the full instrument name at the left side of a staff later than the very beginning of that part.

**Reply by PhysNovalOnGit** [reply]
> I currently can't think of a situation where the full instrument name appears within a part (at the left side of a staff) except at the very beginning.
>For such situations it is clear **how** to display the full instrument name at the beginning if desired.

So are you asking for the MWE of such a layout or were you doubting the appropriateness/necessity of this feature in MusicXML spec and were in the search of typographical oddities that could prove it?

What about [the part of the third oboe/english horn](https://imslp.org/wiki/Special:ReverseLookup/316847) in Ravel's orchestration of the "Pictures at an Exhibition"? Yes, there are only abbrevated names in it, but would you say that this is "a single part (in the MusicXML sense)"?

**Reply by lemzwerg** [reply]
> So are you asking for the MWE of such a layout or were you doubting the appropriateness/necessity of this feature in MusicXML spec and were in the search of typographical oddities that could prove it?

The latter.  I think that `<part-name-display>` was added as a child to `<print>` on purpose – this essentially means that there exists a real-world example that was the reason for the inclusion. @mdgood, do you perhaps remember the details?

> What about [the part of the third oboe/english horn](https://imslp.org/wiki/Special:ReverseLookup/316847) in Ravel's orchestration of the "Pictures at an Exhibition"? Yes, there are only abbrevated names in it, but would you say that this is "a single part (in the MusicXML sense)"?

Yes.  Every movement is a separate part in the MusicXML sense.  In real-world instrument parts, instrument changes in the middle of a piece are *not* indicated by writing out the full name of the changed instrument left of the staff where the change happens.

**Reply by mdgood** [reply, marked as answer]
There are two uses for having `<part-name-display>` as a child of the `<print>` element:

1. In operas and shows with individual numbers and instrument doublings, it is common to want to include all the numbers in a single MusicXML file, at least for a single act if not the whole show. In that case, the part name is displayed at the start of a system midway through the file. See for example the orchestral score of <i>Die Dreigroschenoper</i>. The two woodwind parts double numerous instruments, and the specific instruments are displayed at the start of each number.
2. Music notation programs often allow a part name to be substituted into text. So text on the staff that reads "Alto Saxophone" might be represented as something like "$Part Name$" in the notation program. Specifying the `<part-name-display>` midway through allows the importing program to connect the part name to this text. I expect MNX will come up with a more direct way to make this connection than MusicXML is able to do.

**Reply by lemzwerg** [reply]
Thanks.  My assumption was wrong about a MusicXML part representing a single piece only.  AFAIK, there is not a *single hint* or example in the specification that demonstrates this – how shall I know?

So: Please point me to a MusicXML file that shows how to embed, say, two pieces into a single part.  In particular, I'm interested how measure numbers are 'reset' between pieces.

The second functionality you describe looks very useful to me.  However, as you already conclude, this is again something  I cannot easily derive from reading the specification...
